mino/router/mod.go

```diff
@@ -0,0 +1,53 @@
+// Package router defines the primitives to route a packet among a set of
+// participants.
+package router
+
+import (
+	"go.dedis.ch/dela/mino"
+	"go.dedis.ch/dela/serde"
+)
+
+// Packet is the type of message processed by the router. It contains
+// information that will allow the message to be routed.
+type Packet interface {
+	serde.Message
+
+	// GetSource returns the source address of the message.
+	GetSource() mino.Address
+
+	// GetDestination returns the destination address of the message.
+	GetDestination() []mino.Address
+
+	// GetMessage returns the message to be transmitted to the application when
+	// the source address is the current node. Message is only deserialized when
+	// needed.
```

Comment:
Then return the buffer instead and let mino handles the deserialization, the router doesn't care about it.